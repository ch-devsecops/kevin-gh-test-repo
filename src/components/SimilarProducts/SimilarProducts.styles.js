import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(() =>
  css({
    mt: 'xl',
    mb: ['xl', 'xl', 'big'],
    mx: ['zero', 'zero', 'auto'],
  }),
);

themeStyles.addBaseStyles('Wrapper')(() =>
  css({
    mt: ['xl', 'xl', 'big'],
  }),
);

themeStyles.addBaseStyles('Cell')(() =>
  css({
    display: 'flex',
    pr: 'zero',
    width: '100%',
    justifyContent: 'center',
    mt: ['m', 'default', 'xl'],
  }),
);

themeStyles.addBaseStyles('Header')(() =>
  css({
    textAlign: 'center',
    mb: ['default', 'xl', 'xl'],
    mx: 'auto',
  }),
);

themeStyles.addBaseStyles('CTAButton')(({ theme }) =>
  css({
    [`@media screen and (max-width:${theme.breakpoints[1]})`]: {
      minWidth: '180px',
    },
  }),
);

export default themeStyles;
