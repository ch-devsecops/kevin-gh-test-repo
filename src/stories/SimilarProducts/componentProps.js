import mockData from '../../../sample-mock-data/routes/similar-products/psp_en.json';

export default mockData;
