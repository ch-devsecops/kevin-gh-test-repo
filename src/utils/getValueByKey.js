import { flattenDeep } from 'lodash/array';
/**
 * @param source object to search within
 * @param search key to look for
 * @returns {array}
 */

export default function getValueByKey(source, search) {
  if (!source || typeof source !== 'object') {
    return [];
  }

  const [key, next] = Object.keys(source);
  const { [key]: value, ...rest } = source;

  return flattenDeep([
    ...(key === search ? [value] : getValueByKey(value, search)),
    ...(next ? getValueByKey(rest, search) : []),
  ]);
}
