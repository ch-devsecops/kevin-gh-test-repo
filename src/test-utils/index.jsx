import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
