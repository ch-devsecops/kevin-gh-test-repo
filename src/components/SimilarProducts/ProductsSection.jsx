import React, { useContext, useState } from 'react';
import PropTypes from 'prop-types';
import { SplideSlide } from '@splidejs/react-splide';
import { Image, Optional } from '@honda-canada/design-system-react';
import { useTranslation } from 'react-i18next';

import CarouselSlider from '../CarouselSlider';
import TrimCard from '../TrimCard/TrimCard';
import themeStyles from './SimilarProducts.styles';
import Context from './service/Context';
import ContextTrimPackages from '../TrimPackages/service/Context';
import CTAButton from './shared/CTAButton';
import PriceComponent from '../PriceComponent';
import CompareToggleButton from '../ProductCard/CompareToggleButton';
import { EqualHeight } from '../../utils/components/EqualHeight';
import { ModelExplorationContext } from '../ModelExplorationContext';

const StyledSplide = themeStyles.apply(SplideSlide, 'Cell');

const ProductsSection = ({ relatedItems = [] }) => {
  const { t } = useTranslation();
  const { paymentOptions, vehicleType, isMobile, isFetching, hasError, hydratedTrimsProps, isSmallDesktop } =
    useContext(Context);
  const { navItems } = useContext(ModelExplorationContext) || {};
  const { showAddToCompareButton } = useContext(ContextTrimPackages);
  const [largestTitleHeight, setLargestTitleHeight] = useState(0);
  const modelItemLength = relatedItems?.length;
  const isDesktop = isSmallDesktop ? 3 : 4;
  const ITEMS_PER_PAGE = isMobile ? 1 : isDesktop;
  const trimPath = navItems?.find(item => item.name === 'trims');

  if (modelItemLength === 0) return null;

  const carouselOptions = {
    padding: {
      top: 0,
      left: '8px',
      right: '8px',
    },
    perPage: ITEMS_PER_PAGE,
    perMove: ITEMS_PER_PAGE,
  };

  return (
    <EqualHeight>
      <CarouselSlider
        length={Math.ceil(relatedItems.length / ITEMS_PER_PAGE)}
        splideOptions={carouselOptions}
        hasPagination={relatedItems.length > ITEMS_PER_PAGE}
        paginationControlWithShadow={relatedItems.length > ITEMS_PER_PAGE}
        centerAlign={relatedItems.length < ITEMS_PER_PAGE}
      >
        {() =>
          relatedItems.map(trim => {
            // TODO: Find cleaner implementation to get the modelPathName to form the url
            const pathItems = trimPath?.url?.split('/');
            const modelPathName = pathItems?.[pathItems.length - 1] ?? '';
            const categoryPath = trim?.model?.year?.modelYearPage?.fields?.url ?? '';
            const detailsPath = `${categoryPath}/${modelPathName}?trim=${trim?.trimKey}`;
            const updatedTrimObj = { detailsPath, ...trim };

            return (
              <StyledSplide key={trim.key}>
                <TrimCard
                  ctas={
                    <>
                      <CTAButton trim={updatedTrimObj} />
                      <Optional when={showAddToCompareButton}>
                        <CompareToggleButton detId={trim?.defaultTransmission?.detIdentifier?.value} />
                      </Optional>
                    </>
                  }
                  cursor="pointer"
                  hoverImage={trim.secondaryImage?.src ? <Image {...trim.secondaryImage} /> : null}
                  image={<Image {...trim?.primaryThumbnail?.item?.value} />}
                  language="en"
                  largestTitleHeight={largestTitleHeight}
                  paymentOptions={paymentOptions}
                  priceComponent={
                    <PriceComponent
                      prices={{
                        allInPrice: {
                          value: trim.pricing?.sellingPrice,
                          label: hydratedTrimsProps.sellingPriceLabel,
                        },
                        msrpPrice: {
                          value: trim.pricing?.msrp,
                          label: hydratedTrimsProps.msrpStartingFromLabel,
                        },
                        discount: trim?.pricing?.discount,
                      }}
                      isFetching={isFetching}
                      hasError={hasError}
                      horizontalAlignment="center"
                      height={['auto', 'auto', undefined]}
                      allInPriceTooltipLabel={t('Shared.Common.sellingPriceTooltipLabel')}
                      errorMessage={t('Pages.Models.Exploration.contactLocalDealerLabel')}
                    />
                  }
                  setLargestTitleHeight={setLargestTitleHeight}
                  shouldHidePaymentDetails={false}
                  trim={updatedTrimObj}
                  vehicleType={vehicleType}
                />
              </StyledSplide>
            );
          })
        }
      </CarouselSlider>
    </EqualHeight>
  );
};

ProductsSection.propTypes = {
  relatedItems: PropTypes.arrayOf(PropTypes.shape({})),
};

export default ProductsSection;
