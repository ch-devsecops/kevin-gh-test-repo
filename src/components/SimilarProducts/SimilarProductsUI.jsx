import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { Box, Wrapper as DsrWrapper, H4 } from '@honda-canada/design-system-react';

import ConfigurationProvider from './service/ConfigurationProvider';
import themeStyles from './SimilarProducts.styles';
import { mapGTMCategory } from '../../utils/sitecoreFields';
import Context from './service/Context';
import { getHydratedTrims } from './service/utils';
import ProductsSection from './ProductsSection';

const Container = themeStyles.apply(Box, 'Container');
const Wrapper = themeStyles.apply(DsrWrapper, 'Wrapper');
const Header = themeStyles.apply(H4, 'Header');

export const SimilarProductsUI = ({ relatedItems, componentName, gtmTags }) => {
  const { t } = useTranslation();
  const { setModels, financial, hasError, isFetching, hydratedTrimsProps } = useContext(Context);

  useEffect(() => {
    if (relatedItems?.length) {
      setModels(
        relatedItems.map(item => ({
          modelKey: item.model?.detKey?.value,
          modelYear: item.model?.year?.year?.value,
        })),
      );
    }
  }, [relatedItems?.[0]?.detKey?.value]);

  const hydratedTrims = getHydratedTrims(relatedItems, { financial, hasError, isFetching }, hydratedTrimsProps);

  if (hydratedTrims?.length === 0) return null;

  return (
    <Wrapper data-gtm-category={mapGTMCategory(gtmTags?.gtmCategory)} data-gtm-component-type={componentName}>
      <Header>{t('Pages.Models.Exploration.similarProductsLabel')}</Header>
      <Container>
        <ProductsSection relatedItems={hydratedTrims} />
      </Container>
    </Wrapper>
  );
};

SimilarProductsUI.propTypes = {
  gtmTags: PropTypes.shape({}),
  componentName: PropTypes.string,
  relatedItems: PropTypes.arrayOf(PropTypes.shape({})),
};

const SimilarProducts = ({ appNameConfig, ...restProps }) => (
  <ConfigurationProvider appNameConfig={appNameConfig}>
    <SimilarProductsUI {...restProps} />
  </ConfigurationProvider>
);

SimilarProducts.propTypes = {
  appNameConfig: PropTypes.shape({
    paymentOptions: PropTypes.shape({
      paymentMethod: PropTypes.string,
      paymentFrequency: PropTypes.string,
    }),
    vehicleType: PropTypes.string,
  }),
  gtmTags: PropTypes.shape({}),
  componentName: PropTypes.string,
  relatedItems: PropTypes.arrayOf(PropTypes.shape({})),
};

export default SimilarProducts;
