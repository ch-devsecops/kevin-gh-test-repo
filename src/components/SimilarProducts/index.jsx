import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';

import SimilarProducts from './SimilarProductsUI';
import NotImplemented from '../NotImplemented';
import { PSP_SITE_NAME } from '../../utils/constants';
import { useAppName } from '../../utils/sitecoreContext';
import { variant1 } from './service/constants';

const SimilarRecentlyViewedMap = ({ fields, rendering, ...rest }) => {
  const appName = useAppName();

  if (!fields) return null;

  const mappedFields = fields?.data?.value;
  let relatedItems = [];
  let paymentOptions;
  let fetchFinancial;
  let variant;
  let vehicleType;
  let gtmTags;

  switch (appName) {
    case PSP_SITE_NAME:
      variant = variant1;
      fetchFinancial = false;
      paymentOptions = {
        paymentMethod: 'finance',
        paymentFrequency: 'weekly',
      };

      relatedItems = get(mappedFields, 'relatedTrims.items');
      vehicleType = mappedFields?.relatedTrims?.items?.[0]?.model?.year?.category?.vehicleType?.name;
      gtmTags = {
        gtmTitle: mappedFields.gtmTitle,
        gtmCategory: mappedFields.gtmCategory,
        gtmModelName: mappedFields.gtmModelName,
        gtmTrimName: mappedFields.gtmTrimName,
        gtmBodyStyle: mappedFields.gtmBodyStyle,
        gtmInteractionType: mappedFields.gtmInteractionType,
      };
      break;

    default:
      return <NotImplemented name="SimilarProducts" />;
  }

  return (
    <SimilarProducts
      componentName={rendering?.componentName}
      appNameConfig={{ variant, vehicleType, fetchFinancial, paymentOptions, ...rest }}
      relatedItems={relatedItems}
      gtmTags={gtmTags}
    />
  );
};

SimilarRecentlyViewedMap.propTypes = {
  fields: PropTypes.shape({}),
  rendering: PropTypes.shape({}),
};

export default SimilarRecentlyViewedMap;
