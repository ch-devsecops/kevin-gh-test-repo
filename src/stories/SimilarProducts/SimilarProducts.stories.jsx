import React from 'react';
import LocalStorageContext from '../../components/LocalStorageContext/LocalStorageContext';
import SimilarProducts from '../../components/SimilarProducts';
import mockData from './componentProps';
import sitecoreContext from '../../sitecoreContextMocks/sitecoreContext';

export default {
  title: 'Similar Products PSP',
  component: SimilarProducts,
  parameters: {
    layout: 'fullscreen',
  },
};
const localStorageStateMock = {
  recentlyViewedProducts: ['12094', '12096'],
  toCompareProducts: [],
};

const sitecoreContextMock = sitecoreContext.psp;

const Template = ({ storyProps }) => (
  <LocalStorageContext.Provider value={localStorageStateMock}>
    <SimilarProducts {...storyProps} />
  </LocalStorageContext.Provider>
);

export const Default = Template.bind({});
Default.args = { storyProps: mockData, context: sitecoreContextMock };
