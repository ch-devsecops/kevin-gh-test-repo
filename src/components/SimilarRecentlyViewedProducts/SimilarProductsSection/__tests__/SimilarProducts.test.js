import React from 'react';
import { jssRender, cleanup, axe } from 'test-utils';
import SimilarProducts from '..';
import getValueByKey from '../../../../utils/getValueByKey';

afterEach(cleanup);

const mockData = require('../../../../../sample-mock-data/routes/similar-products/en.json');

const models = getValueByKey(mockData, 'item');

describe('<SimilarProducts>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<SimilarProducts models={models} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should return nulll if there is no  fields', () => {
    const { container } = jssRender(<SimilarProducts models={null} />);

    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<SimilarProducts models={models} />);

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
