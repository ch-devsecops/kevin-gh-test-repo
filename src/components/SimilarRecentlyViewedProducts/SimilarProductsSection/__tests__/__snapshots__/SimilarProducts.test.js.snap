// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimilarProducts> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 hwpdqT"
    display="flex"
    width="1,1,0.5"
  >
    <h4
      aria-label="Pages.Models.Exploration.similarProductsLabel"
      class="sc-1k44dfv-0 chqxmS"
    >
      Pages.Models.Exploration.similarProductsLabel
    </h4>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 heIgvm"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
          data-testid="product-card"
        >
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          />
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          >
            <a
              class="Link-sc-16wd4a0-0 bXBGRT"
              data-gtm-interaction-type="explore"
              data-gtm-trim="GXV160"
              href="https://sitecore.engine.honda.ca/en/GX"
              title="view details"
            >
              <div
                class="sc-1k44dfv-0 cALsOM"
                mb="xs,s"
                mt="0"
              >
                

                <h5
                  class="sc-1k44dfv-0 gXWcpP"
                >
                  GXV160
                </h5>
              </div>
            </a>
          </div>
          <a
            class="Link-sc-16wd4a0-0 bXBGRT"
            data-gtm-interaction-type="explore"
            data-gtm-trim="GXV160"
            href="https://sitecore.engine.honda.ca/en/GX"
            title="view details"
          >
            <div
              class="Box-sc-1bsf784-0 jgcAXQ"
              height="139px,248px"
              order="-1"
            >
              <img
                alt="GX100"
                class="sc-1k44dfv-0 jLTFZk"
                draggable="true"
                loading="lazy"
                src="/sample-mock-data/media/img/product-card/engineImage.png"
                style="object-fit: contain;"
              />
            </div>
          </a>
          <div
            class="Box-sc-1bsf784-0 QoaNQ"
          >
            <div
              class="sc-1k44dfv-0 ktMwmo"
            >
              

              <p>
                Reliable, easy starting and fuel-efficient. There's a reason it's the engine professionals rely on
              </p>
            </div>
          </div>
          <a
            class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
            data-gtm-interaction-type="explore"
            data-gtm-title="view details"
            data-gtm-trim="GXV160"
            data-testid="dsr-button-component"
            display="none,block"
            href="https://sitecore.engine.honda.ca/en/GX"
            tabindex="0"
          >
            Shared.Common.viewDetailsButton
          </a>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
          >
            <label
              class="sc-bwzfXH cvvuMq"
              data-gtm-interaction-type="compare item click"
              data-gtm-title="added comparison item"
              data-gtm-trim="GXV160"
              data-testid="product-card-compare-btn"
              data-testvalue="false"
            >
              <input
                aria-checked="false"
                aria-label="Shared.Common.compareCheckboxAria"
                autocomplete="off"
                class="sc-bdVaJa gBQUrE"
                data-gtm-interaction-type="compare item click"
                data-gtm-title="added comparison item"
                data-gtm-trim="GXV160"
                id="73D22A629A634616B1D171D94246CA91"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="sc-htpNat hVaxDi"
                data-testid="cy-compare-toggle"
              />
              <label
                class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                for="73D22A629A634616B1D171D94246CA91"
              >
                Shared.Common.addToCompareButton
              </label>
            </label>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
          data-testid="product-card"
        >
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          />
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          >
            <a
              class="Link-sc-16wd4a0-0 bXBGRT"
              data-gtm-interaction-type="explore"
              data-gtm-trim="GXV180"
              href="https://sitecore.engine.honda.ca/en/GX"
              title="view details"
            >
              <div
                class="sc-1k44dfv-0 cALsOM"
                mb="xs,s"
                mt="0"
              >
                

                <h5
                  class="sc-1k44dfv-0 gXWcpP"
                >
                  GXV180
                </h5>
              </div>
            </a>
          </div>
          <a
            class="Link-sc-16wd4a0-0 bXBGRT"
            data-gtm-interaction-type="explore"
            data-gtm-trim="GXV180"
            href="https://sitecore.engine.honda.ca/en/GX"
            title="view details"
          >
            <div
              class="Box-sc-1bsf784-0 jgcAXQ"
              height="139px,248px"
              order="-1"
            >
              <img
                alt="GX100"
                class="sc-1k44dfv-0 jLTFZk"
                draggable="true"
                loading="lazy"
                src="/sample-mock-data/media/img/product-card/engineImage.png"
                style="object-fit: contain;"
              />
            </div>
          </a>
          <div
            class="Box-sc-1bsf784-0 QoaNQ"
          >
            <div
              class="sc-1k44dfv-0 ktMwmo"
            >
              

              <p>
                Reliable, easy starting and fuel-efficient. There's a reason it's the engine professionals rely on
              </p>
            </div>
          </div>
          <a
            class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
            data-gtm-interaction-type="explore"
            data-gtm-title="view details"
            data-gtm-trim="GXV180"
            data-testid="dsr-button-component"
            display="none,block"
            href="https://sitecore.engine.honda.ca/en/GX"
            tabindex="0"
          >
            Shared.Common.viewDetailsButton
          </a>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
          >
            <label
              class="sc-bwzfXH cvvuMq"
              data-gtm-interaction-type="compare item click"
              data-gtm-title="added comparison item"
              data-gtm-trim="GXV180"
              data-testid="product-card-compare-btn"
              data-testvalue="false"
            >
              <input
                aria-checked="false"
                aria-label="Shared.Common.compareCheckboxAria"
                autocomplete="off"
                class="sc-bdVaJa gBQUrE"
                data-gtm-interaction-type="compare item click"
                data-gtm-title="added comparison item"
                data-gtm-trim="GXV180"
                id="77510173EDEF4DFBAAF2495E07ECA1C9"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="sc-htpNat hVaxDi"
                data-testid="cy-compare-toggle"
              />
              <label
                class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                for="77510173EDEF4DFBAAF2495E07ECA1C9"
              >
                Shared.Common.addToCompareButton
              </label>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SimilarProducts> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 hwpdqT"
    display="flex"
    width="1,1,0.5"
  >
    <h4
      aria-label="Pages.Models.Exploration.similarProductsLabel"
      class="sc-1k44dfv-0 chqxmS"
    >
      Pages.Models.Exploration.similarProductsLabel
    </h4>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 heIgvm"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
          data-testid="product-card"
        >
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          />
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          >
            <a
              class="Link-sc-16wd4a0-0 bXBGRT"
              data-gtm-interaction-type="explore"
              data-gtm-trim="GXV160"
              href="https://sitecore.engine.honda.ca/en/GX"
              title="view details"
            >
              <div
                class="sc-1k44dfv-0 cALsOM"
                mb="xs,s"
                mt="0"
              >
                

                <h5
                  class="sc-1k44dfv-0 gXWcpP"
                >
                  GXV160
                </h5>
              </div>
            </a>
          </div>
          <a
            class="Link-sc-16wd4a0-0 bXBGRT"
            data-gtm-interaction-type="explore"
            data-gtm-trim="GXV160"
            href="https://sitecore.engine.honda.ca/en/GX"
            title="view details"
          >
            <div
              class="Box-sc-1bsf784-0 jgcAXQ"
              height="139px,248px"
              order="-1"
            >
              <img
                alt="GX100"
                class="sc-1k44dfv-0 jLTFZk"
                draggable="true"
                loading="lazy"
                src="/sample-mock-data/media/img/product-card/engineImage.png"
                style="object-fit: contain;"
              />
            </div>
          </a>
          <div
            class="Box-sc-1bsf784-0 QoaNQ"
          >
            <div
              class="sc-1k44dfv-0 ktMwmo"
            >
              

              <p>
                Reliable, easy starting and fuel-efficient. There's a reason it's the engine professionals rely on
              </p>
            </div>
          </div>
          <a
            class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
            data-gtm-interaction-type="explore"
            data-gtm-title="view details"
            data-gtm-trim="GXV160"
            data-testid="dsr-button-component"
            display="none,block"
            href="https://sitecore.engine.honda.ca/en/GX"
            tabindex="0"
          >
            Shared.Common.viewDetailsButton
          </a>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
          >
            <label
              class="sc-bwzfXH cvvuMq"
              data-gtm-interaction-type="compare item click"
              data-gtm-title="added comparison item"
              data-gtm-trim="GXV160"
              data-testid="product-card-compare-btn"
              data-testvalue="false"
            >
              <input
                aria-checked="false"
                aria-label="Shared.Common.compareCheckboxAria"
                autocomplete="off"
                class="sc-bdVaJa gBQUrE"
                data-gtm-interaction-type="compare item click"
                data-gtm-title="added comparison item"
                data-gtm-trim="GXV160"
                id="73D22A629A634616B1D171D94246CA91"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="sc-htpNat hVaxDi"
                data-testid="cy-compare-toggle"
              />
              <label
                class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                for="73D22A629A634616B1D171D94246CA91"
              >
                Shared.Common.addToCompareButton
              </label>
            </label>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 fZLgxb"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 ljGqou"
          data-testid="product-card"
        >
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          />
          <div
            style="overflow: hidden; transition-property: height; display: block; transition-duration: 0.25s;"
          >
            <a
              class="Link-sc-16wd4a0-0 bXBGRT"
              data-gtm-interaction-type="explore"
              data-gtm-trim="GXV180"
              href="https://sitecore.engine.honda.ca/en/GX"
              title="view details"
            >
              <div
                class="sc-1k44dfv-0 cALsOM"
                mb="xs,s"
                mt="0"
              >
                

                <h5
                  class="sc-1k44dfv-0 gXWcpP"
                >
                  GXV180
                </h5>
              </div>
            </a>
          </div>
          <a
            class="Link-sc-16wd4a0-0 bXBGRT"
            data-gtm-interaction-type="explore"
            data-gtm-trim="GXV180"
            href="https://sitecore.engine.honda.ca/en/GX"
            title="view details"
          >
            <div
              class="Box-sc-1bsf784-0 jgcAXQ"
              height="139px,248px"
              order="-1"
            >
              <img
                alt="GX100"
                class="sc-1k44dfv-0 jLTFZk"
                draggable="true"
                loading="lazy"
                src="/sample-mock-data/media/img/product-card/engineImage.png"
                style="object-fit: contain;"
              />
            </div>
          </a>
          <div
            class="Box-sc-1bsf784-0 QoaNQ"
          >
            <div
              class="sc-1k44dfv-0 ktMwmo"
            >
              

              <p>
                Reliable, easy starting and fuel-efficient. There's a reason it's the engine professionals rely on
              </p>
            </div>
          </div>
          <a
            class="Button-pulw6t-0 sc-1k44dfv-0 eIRfAq"
            data-gtm-interaction-type="explore"
            data-gtm-title="view details"
            data-gtm-trim="GXV180"
            data-testid="dsr-button-component"
            display="none,block"
            href="https://sitecore.engine.honda.ca/en/GX"
            tabindex="0"
          >
            Shared.Common.viewDetailsButton
          </a>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 jkGzMS"
          >
            <label
              class="sc-bwzfXH cvvuMq"
              data-gtm-interaction-type="compare item click"
              data-gtm-title="added comparison item"
              data-gtm-trim="GXV180"
              data-testid="product-card-compare-btn"
              data-testvalue="false"
            >
              <input
                aria-checked="false"
                aria-label="Shared.Common.compareCheckboxAria"
                autocomplete="off"
                class="sc-bdVaJa gBQUrE"
                data-gtm-interaction-type="compare item click"
                data-gtm-title="added comparison item"
                data-gtm-trim="GXV180"
                id="77510173EDEF4DFBAAF2495E07ECA1C9"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="sc-htpNat hVaxDi"
                data-testid="cy-compare-toggle"
              />
              <label
                class="Label-sc-115wxjh-0 sc-1k44dfv-0 jLfsOv"
                for="77510173EDEF4DFBAAF2495E07ECA1C9"
              >
                Shared.Common.addToCompareButton
              </label>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SimilarProducts> should return nulll if there is no  fields 1`] = `<div />`;
